applications/callflow/src/module/cf_voicemail.erl

```diff
@@ -1517,6 +1526,29 @@ store_recording(AttachmentName, DocId, Call) ->
         _Else -> 'false'
     end.
 
+store_recording(AttachmentName, DocId, Call, _Box, 'undefined') ->
+    store_recording(AttachmentName, DocId, Call);
+
+store_recording(AttachmentName, DocId, Call, #mailbox{owner_id=OwnerId}, StorageUrl) ->
+    Url = get_media_url(AttachmentName, DocId, Call, OwnerId, StorageUrl),
+    lager:debug("storing recording ~s at ~s", [AttachmentName, Url]),
+
+    whapps_call_command:b_store(AttachmentName
+                                ,Url
+                                ,Call),
+    timer:sleep(5000),
+
+    case update_doc(<<"external_media_url">>, Url, DocId, Call) of
+        'ok' -> 'true';
+        {'error', _} -> 'false'
+    end.
+
+-spec get_media_url(ne_binary(), ne_binary(), whapps_call:call(), ne_binary(), ne_binary()) -> ne_binary().
+get_media_url(AttachmentName, DocId, Call, OwnerId, StorageUrl) ->
```

Comment:
I've already done this test locally. The `get_message/2` function will be called to resolve the media path. At this moment, if the voicemail metadata contains the `external_media_url` property, it will be returned. The `save_meta/4` was changed to add the `external_media_url` property, if it is present in the media document. 
So, when someone calls the mailbox, Kazoo will send Freeswitch the external URL instead of the resolved CouchDB URL. As Freeswitch will issue an http_get, the external application holding the media will have to handle these requests.